site/docs/spark-procedures.md

```diff
@@ -334,6 +334,10 @@ CALL catalog_name.system.rewrite_manifests('db.sample', false)
 
 The `snapshot` and `migrate` procedures help test and migrate existing Hive or Spark tables to Iceberg.
 
+**Note** Parquet files written with Parquet writers that use names other than `list` and `element` for repeated group
+and element of the list respectively are **read incorrectly** by Iceberg upto 0.12.1 Iceberg versions. Parquet files
+generated by Hive fall in this category.
```

Comment:
Can we try to get a version bound here? Ideally the oss hive version where this starts and stops being an issue?

Reply:
I checked 3.x and it's still called `bag`.

I suggest we consider going with the something similar to the map projection element  rename fix that Ryan applied for maps from Parquet 1.11.0 to 1.11.1, where we don't depend on the name at all.

PR for context: https://github.com/apache/iceberg/pull/3309

It might not work because that was for projection, but if we can get something like that where we don't assume names, that would ideally allow for N levels of nesting etc. cc @SinghAsDev

Reply:
Yea, some details [here](https://github.com/apache/iceberg/pull/3723#issuecomment-992079085)

Reply:
use `!!! Note` for note.

Also "Parquet files generated by Hive" sounds a bit vague. I think we can specify example cases are Parquet files written using (1) `org.apache.hadoop.hive.ql.io.parquet.serde.ParquetHiveSerDe` in Hive, or (2) `spark.sql.parquet.writeLegacyFormat=true` in Spark.